File: new_2024 batch/examInMac.py
def solve():
    n, c = map(int, input().split())

    arr = list(map(int, input().split()))
    print("TEST")
    even = 0
    odd = 0
    if arr[0] != 0:
        even += 1
    if arr[-1] != c:
        if c % 2 == 1:
            odd += 1
        else:
            even += 1
    arr.append(c)
    arr.insert(0, 0)
    print(arr, c)
    for i in range(1, len(arr)):
        if arr[i] - arr[i-1] > 1:
            if (arr[i] - arr[i-1] - 1) % 2 == 1:
                if arr[i-1] % 2 == 0:
                    odd += 1
                else:
                    even += 1
            even += (arr[i] - arr[i-1] - 1) // 2
            odd += (arr[i] - arr[i-1] - 1) // 2

    new_odd = 0
    new_even = 0
    if c % 2 == 1:
        new_even += 1
    new_even += (c // 2)
    new_odd += (c // 2)
    print(odd, even, new_even, new_odd)
    ans = new_odd * odd + new_even * even + (odd * (odd + 1) // 2) + (even * (even + 1) // 2)
    print(ans)

for t in range(int(input())):
    solve()

File: 2025_april/ExamInMac.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

void solve(){
    ll n,c;
    cin >> n >> c;
    vector<int> arr(n);
    // cout << "Hello World";
    for(int i = 0; i < n; i++){
        cin >> arr[i];
    }
    // cout << "Hello";
    ll ans = 0;
    int even = 0;
    int odd = 0;
    for(int i = n-1; i >= 0; i--){
        ans = ans - (arr[i] / 2 + 1);
        ans = ans - (c - arr[i]);
        if (arr[i] % 2){
            ans = ans + (even);
            even += 1;
            
        }else{
            ans = ans + (odd);
            odd += 1;
            
        }

    }

    cout << ans + ((c + 2) * (c + 1)) / 2 << endl;
    
}



int main(){
    int t;
    cin >> t;
    while(t--){
        solve();
    }
}